### ISSUE TITLE
Dendron can't link to notes with non-ascii letters in filename when typing in the name manually

### ISSUE THREAD

**Issue by fnurl** [dendronhq/dendron#2576]
## Please select the area your bug applies to. (Multiple selections are Possible. You can leave blank if you're not sure.)

- [x] Workspace.  VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [ ] Lookup.  Dendron's Lookup Command
- [ ] Views.  Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema.  Dendron Schemas
- [ ] Pod.  Data import from / export to Dendron
- [ ] Publish.  External Site Publish
- [ ] Markdown.  Markdown features, such as syntax support and features

## Describe the bug

Dendron has no problems in creating notes with non-ascii letters from lookup or when typing `ctrl+enter` when the cursor is on a wikilink. Example note name `värde.returvärde`. If the name of the note is typed in a wikilink manually, i.e. not selected from the popup, Dendron fails to recognize that the note name is an existing note, i.e. does not link to the note.

I also believe that the popup window disappearing when typing a non-ascii letter is related to this bug.

## To Reproduce

1. Create a new note with the name `värde.returvärde`
2. Link to note only by typing in the text, i.e. [[värde.returvärde]]

## Expected behavior

Note name is recognized and linked.

## Desktop (please complete the following information)

- OS: macOS
- Versions (all that apply to bug):
  - Dendron Extension: 0.85.0
  - VS Code version: 1.65.2

**Comment by kevinslin** [comment]
hmm, not able to reproduce. what version of dendron are you using?

**Comment by fnurl** [comment]
This is weird. I created a new dendron workspace and tried this there and it works. I recorded a video showing the behavior in my existing workspace and it working in a new one. I also tried refactoring the hierarchy in my current workspace replacing `värde` with `value`, but the preview comes up empty even though there are multiple notes with filenames that start with `värde`.

**Comment by fnurl** [comment]
I might have a lead. Do you normalize utf-8 strings? 

There are two ways of representing the letter ä, the first is to use a single unicode codepoint; Latin Small Letter a with Diaeresis (`U+00E4`), the second is to use the letter a and the Combining Diaeresis ( `U+0308`). There might be some extra implementation details, but that's the gist of it. So I might be experiencing my typed letter `ä` having a different underlying representation compared to the string read from the filesystem.

I know that e.g. Python automatically normalizes unicode strings, but it might be the case that it needs to be done explicitly in VS Code? (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize)

Though I have not renamed the files outside of VS Code, they are stored in my Dropbox something might have happened there. I may have also have renamed files inside VS Code (after creating them via the lookup command).  Also, I am running macOS 12.1 on a APFS drive which in contrast to HFS+ does not normalize file names (see e.g. https://eclecticlight.co/2017/04/06/apfs-is-currently-unusable-with-most-non-english-languages/).

I guess one workaround is for me to rename my notes and not use non-ascii letters :) Still might be something to look into for Dendron though?

**Comment by fnurl** [comment]
Yeah, definitely normalization errors. It looks like VS Code does not normalize:

https://www.dropbox.com/s/zm4dgk9d69y1e97/not-found-normalization-error.png?dl=0

I did open the vault in Obsidian and Logseq to try them a bit. Might be the case that one of them changed the encoding used in my files too.

**Comment by fnurl** [comment]
I think this is the final comment here. :)

After renaming my files I have been going through my note content and manually fixing the links. I noticed that it was only the occurrences of the letter `ä` in wikilinks that had been changed to `a + ¨` since pressing backspace with the cursor directly after the letter `ä` in a wikilink changed the character to an `a` which means that there was a diaeresis character after the `a`.

As this only happened for the `ä` characters in the wikilinks, and not the the `ä` characters in my normal body text, the string error most likely came from the file system.. so if you do introduce string normalization for my case my keyboard enters the composed characters (NFC), not the decomposed characters (NFD).

**Comment by fnurl** [comment]
@kevinslin I did some tests and I have a suggestion :)

It is Dropbox performing unicode normalizations on the filenames. Here are the results from creating two separate Dendron workspaces, one in a local only folder and the other in a folder in my Dropbox. I then created the same notes with non-ascii unicode names using the lookup command wikilinked to them, quit VS code, then opened VS Code and linked to the files again.

![image](https://user-images.githubusercontent.com/106794/159096064-b26c22c3-d874-4d47-8aa9-ae1227fb0ffc.png)

I created also script to test the renaming and Dropbox changes the filenames at once (I guess when the file is no longer locked). See results (in comment) and script here: https://gist.github.com/fnurl/e91080b2540b1cf0c3ac29adbf75ce26

**Suggestion** I think a solution would be for Dendron to always normalize any note name it handles. No change has to be made to the files that are saved, just the strings with the note names, whether they are read from the filesystem or extracted from some note text. That way, no matter what representation is used in the filesystem or in the note text file, Dendron will always be able to correctly link a wikilink to its note.

Not normalizing note names internally can negatively affect any user that meets the criteria:

- uses a APFS file system
- uses non-ascii filenames
- stores the Dendron vault in a Dropbox folder (and perhaps other sync services)

Have a nice Friday!
